packages/juce-blueprint/esm/Blueprint.js

```diff
@@ -22,52 +22,85 @@ export function Image(props) {
   return React.createElement('Image', props, props.children);
 }
 
-export function Canvas(props) {
-  if (props.hasOwnProperty('onDraw')) {
-    let userOnDraw = props.onDraw;
-    let wrappedDraw = function(ctx) {
-      Object.defineProperty(ctx, 'fillStyle', {
-        enumerable: false,
-        configurable: false,
-        get: function() {
-          return 'Not Supported';
-        },
-        set: function(value) {
-          this.__setFillStyle(value);
-        }
-      });
-
-      Object.defineProperty(ctx, 'strokeStyle', {
-        enumerable: false,
-        configurable: false,
-        get: function() {
-          return 'Not Supported';
-        },
-        set: function(value) {
-          this.__setStrokeStyle(value);
-        }
-      });
-
-      Object.defineProperty(ctx, 'lineWidth', {
-        enumerable: false,
-        configurable: false,
-        get: function() {
-          return 'Not Supported';
-        },
-        set: function(value) {
-          this.__setLineWidth(value);
-        }
-      });
-
-      return userOnDraw(ctx);
+export function bindCanvasContextProperties(ctx) {
```

Comment:
:+1: definitely. This `export` here marks this method as part of the "public" blueprint js API, and I'm not sure we want this to be publicly accessible, but we definitely want it to be shared and shareable internally.

But, let's not worry about that for now. Once the native side of blueprint is more complete (i.e. as we approach a "beta" release or something) we will want to take a pass at what our js API should look like

Reply:
Yeah. I basically have an initial concept I'm messing around with for the look and feel override stuff. It's not a pattern I would advocate for so much now I've dug into it but it's something my current shop wants me to investigate to allow us to get up and running fast. 

```
function ReactBasedSlider(props) {
    if (props.hasOwnProperty('drawLinearSlider')) {
        let userDrawLinearSlider = props.drawLinearSlider;
        let wrappedDraw = function(ctx, width, height, sliderPos, sliderStyle) {
            bindCanvasContextProperties(ctx);
            return userDrawLinearSlider(ctx, width, height, sliderPos, sliderStyle);
        };
        return React.createElement('ReactBasedSlider', Object.assign({}, props, {
            drawLinearSlider: wrappedDraw,
        }), props.children);
    }

    return React.createElement('ReactBasedSlider', props, props.children);
}

export default ReactBasedSlider;
```